lightning/src/ln/channel.rs

```diff
@@ -3771,6 +3810,72 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 		self.channel_transaction_parameters.channel_type_features = self.channel_type.clone();
 		Ok(())
 	}
+
+	// /// Splice process starting; update state; update capacity, state, reset funding tx
+	// #[cfg(splicing)]
+	// pub(crate) fn splice_start<L: Deref>(&mut self, is_outgoing: bool, logger: &L) where L::Target: Logger {
+	// 	// Set state, by this point handshake is complete
+	// 	self.channel_state = ChannelState::NegotiatingFunding(NegotiatingFundingFlags::OUR_INIT_SENT | NegotiatingFundingFlags::THEIR_INIT_SENT);
+
+	// 	log_info!(logger,
+	// 		"Splicing process started, new channel value {}, outgoing {}, channel_id {}",
+	// 		self.channel_value_satoshis, is_outgoing, self.channel_id,
+	// 	);
+	// }
+
+	/// Get the splice message that can be sent during splice initiation.
+	#[cfg(splicing)]
+	pub fn get_splice_init(&self, our_funding_contribution_satoshis: i64, signer_provider: &SP,
+		funding_feerate_perkw: u32, locktime: u32,
+	) -> msgs::SpliceInit {
+		if !self.is_outbound() {
+			panic!("Tried to initiate a splice on an inbound channel!");
+		}
```

Comment:
Why is this disallowed?

Reply:
Good point, the splice initiator can be any of the channel peers (probably copy-paste from regular channel open)